mteb/benchmarks/benchmarks.py

```diff
@@ -3,16 +3,34 @@
 from dataclasses import dataclass
 from typing import Sequence
 
+from pydantic import AnyUrl, BeforeValidator, TypeAdapter
+from typing_extensions import Annotated
+
 from mteb.abstasks.AbsTask import AbsTask
 from mteb.overview import get_tasks
 
+http_url_adapter = TypeAdapter(AnyUrl)
+STR_URL = Annotated[
+    str, BeforeValidator(lambda value: str(http_url_adapter.validate_python(value)))
+]  # Allows the type to be a string, but ensures that the string is a URL
+
 
 @dataclass
 class Benchmark:
+    """A benchmark object intended to run certain a full benchmark within MTEB.
+
+    Args:
+        name: The name of the benchmark
+        tasks: The tasks within the benchmark.
+        description: A description of the benchmark, should include its intended goal and potentially a description of its construction
+        reference: A link reference, to a source containing additional information typically to a paper, leaderboard or github.
+        citation: A bibtex citation
+    """
+
     name: str
-    tasks: Sequence[str] | Sequence[AbsTask]
+    tasks: Sequence[AbsTask]
```

Comment:
Maybe it would make sense to add a `classmethod` that allows users to create a benchmark from task names then for convenience. (very optional)

Reply:
+1. Not sure if this breaks backwards compatibility of the Benchmark object.

Reply:
Something like:
```
Benchmark.from_task_names([...])
```
Isn't that just `mteb.get_tasks(tasks=[...])`, can def. implement it, but not sure it is worth it. Otherwise, I might be misunderstanding something?

We can maintain backward compatibility by creating an init method that validates the tasks but keeps them as strings. However, I don't believe this interface has been used that much so I wouldn't worry too much about changing it and I do think that using task names instead of tasks leads to more bugs (e.g. for two of the benchmarks the tasks were invalid, and for MTEB(eng) the task the benchmark included non-English tasks).

Reply:
>  I don't believe this interface has been used that much so I wouldn't worry too much about changing it and I do think that using task names instead of tasks leads to more bugs 

That's fair. I'm ok with not implementing "from_task_names". Can we instead add an example in the docstrings to include `mteb.get_tasks(tasks=[...])` if users do want to use strings to choose tasks?

Reply:
Added an example :)